### PR TITLE
PR 05 - User can create account

### PR DESCRIPTION
**Pull request by ghost**
```
As a User,
In order to have a Playdate user profile,
I would like to be able to create an account.
```
Upon creating an account (sign up), users will associate kindergartens with their account to be able to create playdates, and see playdates, that include children from those kindergartens.  This PR creates the User model using the Devise gem to enable in further PRs,
1. Associate Kindergarten model --> User model has many Kindergartens
2. To set permissions to ensure that users can only post and see playdates associated with their own kindergartens.

[Link to pivital tracker](https://www.pivotaltracker.com/story/show/167340498)

### CODE REVIEW

File: spec/models/user_spec.rb [hunk 1]
```diff
@@ -0,0 +1,5 @@
+require 'rails_helper'
+
+RSpec.describe User, type: :model do
+  pending "add some examples to (or delete) #{__FILE__}"
+end
```
Comment (tochman) [inline spec/models/user_spec.rb:19]: No specs?

File: spec/factories/users.rb [hunk 2]
```diff
@@ -0,0 +1,5 @@
+FactoryBot.define do
+  factory :user do
+    
+  end
+end
```
Comment (tochman) [inline spec/factories/users.rb:7]: No attributes in the factory?

**Review by tochman — CHANGES_REQUESTED** [review]
I can't really make the connection between the presented implementation and `to see and make playdates` part of the user story. 

There's no need to generate the devise views at this point

File: features/step_definitions/basic_steps.rb [hunk 3]
```diff
@@ -1,3 +1,11 @@
...
+Given("I visit the dashboard") do
+  visit root_path
+end
+
+When("I fill {string} with {string}") do |field, content|
+  fill_in field, with: content
+end
+
+When("I click {string}") do |name|
+  click_link_or_button(name)
```
Comment (oliverochman) [inline features/step_definitions/basic_steps.rb]: please write this line in poetry mode

**Comment by tochman** [comment]
I understand the `This PR creates the User model using the Devise gem to enable in further PRs,` part, but I don't see a reason for implement authentication at this point unless there is any use for that.

File: spec/models/user_spec.rb [hunk 4]
```diff
@@ -0,0 +1,5 @@
+require 'rails_helper'
+
+RSpec.describe User, type: :model do
+  pending "add some examples to (or delete) #{__FILE__}"
```
Comment (oliverochman) [inline spec/models/user_spec.rb]: You need to unit test the User model

**Review by oliverochman — CHANGES_REQUESTED** [review]

File: spec/factories/users.rb (hunk 2 above)
Comment (dashlykova) [reply spec/factories/users.rb:7]: We have addressed your insights. Created Factory Bot with email and password.

File: spec/models/user_spec.rb (hunk 1 above)
Comment (dashlykova) [reply spec/models/user_spec.rb:19]: We have added the unit tests now.

File: spec/models/user_spec.rb (hunk 4 above)
Comment (ghost) [reply spec/models/user_spec.rb]: Added.

File: features/step_definitions/basic_steps.rb (hunk 3 above)
Comment (ghost) [reply features/step_definitions/basic_steps.rb]: Fixed.

**Review by tochman — CHANGES_REQUESTED** [review]
Please fix code formatting in the feature files to be consistent with the rest of the files.

Regarding the change to the user story once you started development. It is okay now, but it would never pass a code review in a real setting.

**Comment by tochman** [comment]
I still don't understand why you keep the Devise i18 files and the views (since you are not making any changes to them)  but I already stated my case.

File: app/views/devise/registrations/new.html.erb [hunk 5]
```diff
@@ -0,0 +1,29 @@
+<h2>Sign up</h2>
```
Comment (oliverochman) [inline app/views/devise/registrations/new.html.erb]: Are you adding anything to these files?
If not, please remove them

File: app/views/devise/shared/_error_messages.html.erb [hunk 6]
```diff
@@ -0,0 +1,15 @@
+<% if resource.errors.any? %>
```
Comment (oliverochman) [inline app/views/devise/shared/_error_messages.html.erb]: same here

File: app/views/devise/shared/_links.html.erb [hunk 7]
```diff
@@ -0,0 +1,25 @@
+<%- if controller_name != 'sessions' %>
```
Comment (oliverochman) [inline app/views/devise/shared/_links.html.erb]: same here

File: app/views/layouts/application.html.haml [hunk 8]
```diff
@@ -2,7 +2,7 @@
 %html
   %head
     %meta{content: 'text/html; charset=UTF-8', http: {equiv: "Content-Type"}}/
-    %title Kibbles n' Bites - Fine Pet Cuisine
+    %title PlayDate
```
Comment (oliverochman) [inline app/views/layouts/application.html.haml:5]: If you get problems with displaying flashes when removing the devise views, then you need to display flashes in this file

**Review by oliverochman — CHANGES_REQUESTED** [review]

File: app/views/layouts/application.html.haml (hunk 8 above)
Comment (ghost) [reply app/views/layouts/application.html.haml:5]: Understood.

File: app/views/devise/shared/_links.html.erb (hunk 7 above)
Comment (ghost) [reply app/views/devise/shared/_links.html.erb]: Removed.

File: app/views/devise/shared/_error_messages.html.erb (hunk 6 above)
Comment (ghost) [reply app/views/devise/shared/_error_messages.html.erb]: Removed.

File: app/views/devise/registrations/new.html.erb (hunk 5 above)
Comment (ghost) [reply app/views/devise/registrations/new.html.erb]: Removed.

**Review by oliverochman — APPROVED** [review]

**Pull request merged**